scipy/stats/_continuous_distns.py

```
@@ -8603,6 +8925,8 @@ class halfgennorm_gen(rv_continuous):
     %(example)s
 
     """
+    def _shape_info(self):
+        return [_ShapeInfo("beta", False, (0, np.inf), (False, False))]
```

This distribution seems to work without an `_argcheck` even though `beta` is constrained. Is it because the functions naturally return `nan`? Should there be an `_argcheck` anyway, for good measure?

(Or should `_argcheck` have a default implementation based on `_shape_info` that can be overridden in cases where one parameter's bounds depends on the bounds of another?)

Not really part of this PR, but it is a natural question to ask while reviewing.

---

The default `_argcheck` of `rv_continuous` masks outputs with nan for negative shape parameters. So, `halfgennorm` works with a `_argcheck` is doesn't require one too.

See this: https://github.com/scipy/scipy/blob/4d66d813205fc7280633469b3e2a7276b4dfe72e/scipy/stats/_distn_infrastructure.py#L959-L969

---

Thanks! I ended up discovering that after seeing a few of these.